hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/ReflectiveCompactSerializer.java

```diff
@@ -0,0 +1,444 @@
+ */
+
+package com.hazelcast.internal.serialization.impl.compact;
+
+import com.hazelcast.internal.nio.ClassLoaderUtil;
+import com.hazelcast.internal.util.ExceptionUtil;
+import com.hazelcast.nio.serialization.FieldType;
+import com.hazelcast.nio.serialization.compact.CompactReader;
+import com.hazelcast.nio.serialization.compact.CompactWriter;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.hazelcast.nio.serialization.FieldType.BOOLEAN;
+import static com.hazelcast.nio.serialization.FieldType.BOOLEAN_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.BYTE;
+import static com.hazelcast.nio.serialization.FieldType.BYTE_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.CHAR;
+import static com.hazelcast.nio.serialization.FieldType.CHAR_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.COMPOSED;
+import static com.hazelcast.nio.serialization.FieldType.COMPOSED_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.DATE;
+import static com.hazelcast.nio.serialization.FieldType.DATE_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.DECIMAL;
+import static com.hazelcast.nio.serialization.FieldType.DECIMAL_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.DOUBLE;
+import static com.hazelcast.nio.serialization.FieldType.DOUBLE_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.FLOAT;
+import static com.hazelcast.nio.serialization.FieldType.FLOAT_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.INT;
+import static com.hazelcast.nio.serialization.FieldType.INT_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.LONG;
+import static com.hazelcast.nio.serialization.FieldType.LONG_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.SHORT;
+import static com.hazelcast.nio.serialization.FieldType.SHORT_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.TIME;
+import static com.hazelcast.nio.serialization.FieldType.TIMESTAMP;
+import static com.hazelcast.nio.serialization.FieldType.TIMESTAMP_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.TIMESTAMP_WITH_TIMEZONE;
+import static com.hazelcast.nio.serialization.FieldType.TIMESTAMP_WITH_TIMEZONE_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.TIME_ARRAY;
+import static com.hazelcast.nio.serialization.FieldType.UTF;
+import static com.hazelcast.nio.serialization.FieldType.UTF_ARRAY;
+import static java.util.stream.Collectors.toList;
+
+public class ReflectiveCompactSerializer implements InternalCompactSerializer<Object, DefaultCompactReader> {
```

Comment:
Where did we document the limitations of this class, e.g. no support for enum, limitations of Java reflection, e.g.lists, etc.?

Reply:
We don't have it yet. I can document it here, but we need a public documentation as well. Preferably on the reference manual.

Reply:
I am adding a javadoc to both ReflectiveCompactSerializer and also CompactSerializationConfig